pkgs/misc/uboot/default.nix

```diff
@@ -323,6 +323,18 @@ in {
     filesToInstall = ["u-boot.bin"];
   };
 
+  ubootQemuRiscv64Smode = buildUBoot {
+    defconfig = "qemu-riscv64_smode_defconfig";
+    extraMeta.platforms = ["riscv64-linux"];
+    extraPatches = [
+      (fetchpatch {
+        url = "https://salsa.debian.org/debian/u-boot/-/raw/sid/debian/patches/riscv64/qemu-riscv64_smode-sifive-fu540-fix-extlinux-define-.patch";
+        sha256 = "0vf9v8zd6k1hlglfk29mzcsz9dxsfp112llhwx627wzbsy6hrcgm";
+      })
```

Comment:
Can we fetch from the original source?

https://patchwork.ozlabs.org/project/uboot/patch/20190821190720.4286-1-david.abdurachmanov@sifive.com/

The `mbox` ilnk is what can be applied:
→ https://patchwork.ozlabs.org/project/uboot/patch/20190821190720.4286-1-david.abdurachmanov@sifive.com/mbox/

In *theory* the hash should stay the same, since patches are normalized.

Reply:
There is an extra hunk that cannot be applied. Perhaps we can move the patch in-tree as I'm not sure if it's a good idea to hotlink to Debian's semi-internal developer infra.

Reply:
Right, I see there's one less hunk.

This is permalink'd to the current commit:

 - https://salsa.debian.org/debian/u-boot/-/raw/1e3ecdc4e42a566671ba24e9b56ecf465be7bd40/debian/patches/riscv64/qemu-riscv64_smode-sifive-fu540-fix-extlinux-define-.patch

Reply:
Ok, updated the commit to pin to the correct commit.